includes/Checker/Checks/Plugin_Readme_Check.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\Checks\Plugin_Readme_Check
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker\Checks;
+
+use WordPress\Plugin_Check\Checker\Check_Result;
+
+/**
+ * Check the plugins readme.txt file and contents.
+ *
+ * @since n.e.x.t
+ */
+class Plugin_Readme_Check extends Abstract_File_Check {
+
+	/**
+	 * Check the readme.txt file.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The Check Result to amend.
+	 * @param array        $files  Array of plugin files.
+	 */
+	protected function check_files( Check_Result $result, array $files ) {
+		// Find the readme file.
+		$readme = self::filter_files_by_regex( $files, '/readme\.txt/' );
+
+		// If the readme.txt does not exist, add a warning and skip other tests.
+		if ( empty( $readme ) ) {
+			$result->add_message(
+				false,
+				__( 'The plugins readme.txt does not exist.', 'plugin-check' ),
+				array(
+					'file' => 'readme.txt',
+					'code' => 'plugin_readme.does_not_exist',
+				)
+			);
+
+			return;
+		}
+
+		// Check the readme.txt for default text.
+		$this->readme_check_default_text( $result, $readme );
+
+		// Check the readme.txt for a valid license.
+		$this->readme_check_license( $result, $readme );
+
+		// Check the readme.txt for a valid version.
+		$this->readme_check_version( $result, $readme );
+	}
+
+	/**
+	 * Checks the readme.txt for default text.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The Check Result to amend.
+	 * @param array        $files  Array of plugin files.
+	 */
+	private function readme_check_default_text( Check_Result $result, array $files ) {
+		if ( self::file_str_contains( $files, 'Here is a short description of the plugin.' ) ) {
+			$result->add_message(
+				false,
+				__( 'The plugins readme.txt appears to contain default text.', 'plugin-check' ),
+				array(
+					'code' => 'plugin_readme.contains_default_text',
+					'file' => $result->plugin()->path( '/readme.txt' ),
+				)
+			);
+		}
+	}
+
+	/**
+	 * Checks the readme.txt for a valid license.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The Check Result to amend.
+	 * @param array        $files  Array of plugin files.
+	 */
+	private function readme_check_license( Check_Result $result, array $files ) {
+		if (
+			! self::file_str_contains( $files, 'License:' ) ||
+			! self::file_str_contains( $files, 'License URI:' )
```

Comment:
These checks aren't sufficient to check the _validity_ of the license, they only check for the existence of any license. Please see https://github.com/WordPress/plugin-check/blob/trunk/checks/readme.php#L31.

I think it would be more efficient to use that regex and combine it with the `(License:|License URI:)` bit so that you can then rely on a single call to `self::file_preg_match()` to check if the readme is valid or not.

Reply:
Thanks @felixarntz I've updated this with a regex rule to match that used in the other plugin check